src/io/flutter/pub/PubRoot.java

```diff
@@ -126,8 +126,14 @@ public static PubRoot forEventWithRefresh(@NotNull final AnActionEvent event) {
    * Based on the filesystem cache; doesn't refresh anything.
    */
   @Nullable
-  public static PubRoot forPsiFile(@NotNull PsiFile file) {
-    return forDescendant(file.getVirtualFile(), file.getProject());
+  public static PubRoot forPsiFile(@NotNull PsiFile psiFile) {
+    final VirtualFile file = psiFile.getVirtualFile();
+    if (isPubspec(file)) {
```

Comment:
Looking at `PsiFileImpl`, I see `getVirtualFile()` may technically return `null`. Though I'm not sure how likely it would be in practice, maybe consider guarding all the same?